README.md

```diff
@@ -33,108 +34,242 @@ Menu
+- Set the total number of script iterations with the `--iterations`/`-i` CLI flag (or the `K6_ITERATIONS` environment variable and the `iterations` script/JSON option). k6 will stop executing the script whenever the **total** number of iterations (i.e. the number of iterations across all VUs) reaches the specified number. So if you have `k6 run --iterations 10 --vus 10 script.js`, then each VU would make only a single iteration.
+- Set both the test duration and the total number of script iterations. In that case, k6 would stop the script execution whenever either one of the above conditions is reached first.
+
+For more complex cases, you can specify execution stages. They are a combination of `duration,target-VUs` pairs that tell k6 the number of VUs you want the script execution to ramp up or ramp down to over which period. You can set them via the `stages` script/JSON option as an array of `{ duration: ..., target: ... }` pairs, or with the `--stage`/`-s` CLI flags and the `K6_STAGE` environment variable via the `duration:target,duration:target...` syntax.
+
+For example, the following options would have k6 linearly ramping up from 5 to 10 VUs over the period of 3 minutes (k6 starts with `vus` number of VUs, or 1 by default), then staying flat at 10 VUs for 5 minutes, then ramping up from 10 to 35 VUs over the next 10 minutes before finally ramping down to 0 VUs for another 90 seconds.
+
+```js
+export let options = {
+    vus: 5,
+    stages: [
+        { duration: "3m", target: 10 },
+        { duration: "5m", target: 10 },
+        { duration: "10m", target: 35 },
+        { duration: "1m30s", target: 0 },
+    ]
 };
 ```
+Alternatively, you can use the CLI flags `--vus 5 --stage 3m:10,5m:10,10m:35,1m30s:0` or set the environment variables `K6_VUS=5 K6_STAGE="3m:10,5m:10,10m:35,1m30s:0"` to achieve the same results.
 
-Save it as `script.js`, then run k6 like this:
+For a complete list of supported k6 options, take a look at the documentation at [https://docs.k6.io/docs/options](https://docs.k6.io/docs/options).
 
-`k6 run script.js`
+_Hint: besides accessing the supplied [environment variables](https://docs.k6.io/docs/environment-variables) through the `__ENV` global object briefly mentioned above, you can also use the [execution context variables](https://docs.k6.io/docs/execution-context-variables) `__VU` and `__ITER` to access the current VU number and the number  oof the current iteration **in that VU**. These variables can be very useful if you want to have different scenarios for different VUs or you want to generate different data from different VUs: ```http.post("https://some.example.website/signup", {username: `testuser${__VU}@testsite.com`, /* ... */})```_
```

Comment:
_Hint: besides accessing the supplied [environment variables](https://docs.k6.io/docs/environment-variables) through the `__ENV` global object briefly mentioned above, you can also use the [execution context variables](https://docs.k6.io/docs/execution-context-variables) `__VU` and `__ITER` to access the current VU number and the number of the current iteration **for that VU**. These variables can be very useful if you want VUs to execute different scripts/scenarios or to aid in generating different data per VU. ```http.post("https://some.example.website/signup", {username: `testuser${__VU}@testsite.com`, /* ... */})```_